server/runtime_nakama_module.go

```diff
@@ -0,0 +1,1108 @@
+	}
+
+	expr, err := cronexpr.Parse(cron)
+	if err != nil {
+		l.ArgError(1, "expects a valid cron string")
+		return 0
+	}
+	t := time.Unix(ts, 0).UTC()
+	next := expr.Next(t)
+	nextTs := next.UTC().Unix()
+	l.Push(lua.LNumber(nextTs))
+	return 1
+}
+
+func (n *NakamaModule) loggerInfo(l *lua.LState) int {
+	message := l.CheckString(1)
+	if message == "" {
+		l.ArgError(1, "expects message string")
+		return 0
+	}
+	n.logger.Info(message)
+	l.Push(lua.LString(message))
+	return 1
+}
+
+func (n *NakamaModule) loggerWarn(l *lua.LState) int {
+	message := l.CheckString(1)
+	if message == "" {
+		l.ArgError(1, "expects message string")
+		return 0
+	}
+	n.logger.Warn(message)
+	l.Push(lua.LString(message))
+	return 1
+}
+
+func (n *NakamaModule) loggerError(l *lua.LState) int {
+	message := l.CheckString(1)
+	if message == "" {
+		l.ArgError(1, "expects message string")
+		return 0
+	}
+	n.logger.Error(message)
+	l.Push(lua.LString(message))
+	return 1
+}
+
+func (n *NakamaModule) streamUserGet(l *lua.LState) int {
+	// Parse input User ID.
+	userIDString := l.CheckString(1)
+	if userIDString == "" {
+		l.ArgError(1, "expects user id")
+		return 0
+	}
+	userID, err := uuid.FromString(userIDString)
+	if err != nil {
+		l.ArgError(1, "expects valid user id")
+		return 0
+	}
+
+	// Parse input Session ID.
+	sessionIDString := l.CheckString(2)
+	if sessionIDString == "" {
+		l.ArgError(2, "expects session id")
+		return 0
+	}
+	sessionID, err := uuid.FromString(sessionIDString)
+	if err != nil {
+		l.ArgError(2, "expects valid session id")
+		return 0
+	}
+
+	// Parse input stream identifier.
+	streamTable := l.CheckTable(3)
+	if streamTable == nil || streamTable.Len() == 0 {
+		l.ArgError(3, "expects a valid stream")
+		return 0
+	}
+	stream := PresenceStream{}
+	conversionError := ""
+	streamTable.ForEach(func(k lua.LValue, v lua.LValue) {
+		switch k.String() {
+		case "Mode":
+			if v.Type() != lua.LTNumber {
+				conversionError = "stream Mode must be a number"
+				return
+			}
+			stream.Mode = uint8(lua.LVAsNumber(v))
+		case "Subject":
+			if v.Type() != lua.LTString {
+				conversionError = "stream Subject must be a string"
+				return
+			}
+			sid, err := uuid.FromString(v.String())
+			if err != nil {
+				conversionError = "stream Subject must be a valid identifier"
+				return
+			}
+			stream.Subject = sid
+		case "Descriptor":
+			if v.Type() != lua.LTString {
+				conversionError = "stream Descriptor must be a string"
+				return
+			}
+			did, err := uuid.FromString(v.String())
+			if err != nil {
+				conversionError = "stream Descriptor must be a valid identifier"
+				return
+			}
+			stream.Subject = did
+		case "Label":
+			if v.Type() != lua.LTString {
+				conversionError = "stream Label must be a string"
+				return
+			}
+			stream.Label = v.String()
+		}
+	})
+	if conversionError != "" {
+		l.ArgError(3, conversionError)
+		return 0
+	}
+
+	meta := n.tracker.GetLocalBySessionIDStreamUserID(sessionID, stream, userID)
```

Comment:
Does this code need changing in NE version? Different tracker call?

Reply:
If so, maybe worth having a func call that looks at both Local and cluster presences?

Reply:
Stream presences are checked/mutated on the node they belong to only, so this is correct behaviour until that changes.